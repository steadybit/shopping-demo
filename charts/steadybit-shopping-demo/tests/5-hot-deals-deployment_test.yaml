templates:
  - 5-hot-deals-deployment.yml
chart:
  appVersion: v0.0.0
tests:
  - it: manifest should match snapshot
    asserts:
      - matchSnapshot: { }
  - it: manifest should match snapshot replica count
    set:
      hotDeals:
        replicaCount: 2
    asserts:
      - matchSnapshot: { }
  - it: manifest should match snapshot with podAffinity and service type and podDisruptionBudget
    set:
      hotDeals:
        replicaCount: 2
        podAffinity: true
        service:
          type: ClusterIP
        podDisruptionBudget: true
    asserts:
      - matchSnapshot: { }