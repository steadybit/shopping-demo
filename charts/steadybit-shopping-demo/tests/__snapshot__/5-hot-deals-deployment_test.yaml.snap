manifest should match snapshot:
  1: |
    apiVersion: v1
    data:
      nginx.conf: |-
        user  nginx;
        worker_processes  1;

        error_log  /var/log/nginx/error.log warn;
        pid        /var/run/nginx.pid;

        events {
            worker_connections  1024;
        }

        http {
            include       /etc/nginx/mime.types;
            default_type  application/octet-stream;

            access_log off;

            sendfile        on;
            keepalive_timeout  65;

            upstream hot-deals {
                server 127.0.0.1:8099;
            }

            server {
                listen 8083;

                location / {
                    proxy_pass         http://hot-deals;
                    proxy_redirect     off;
                }
            }
        }
    kind: ConfigMap
    metadata:
      name: hotdeals-nginx-conf
      namespace: NAMESPACE
  2: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        domain: shop-products
        run: hot-deals
        service-tier: "2"
        tags.datadoghq.com/service: shop-products
        tags.datadoghq.com/version: 1.0.0
        tags.steadybit.com/service-validation: http
      name: hot-deals
      namespace: NAMESPACE
    spec:
      replicas: 1
      selector:
        matchLabels:
          run: hot-deals-exposed
      strategy:
        rollingUpdate:
          maxSurge: 0
          maxUnavailable: 1
        type: RollingUpdate
      template:
        metadata:
          labels:
            domain: shop-products
            run: hot-deals-exposed
            service-tier: "2"
            tags.datadoghq.com/service: shop-products
            tags.datadoghq.com/version: 1.0.0
            tags.steadybit.com/service-validation: http
        spec:
          containers:
            - env:
                - name: SERVER_PORT
                  value: "8099"
                - name: DD_SERVICE
                  valueFrom:
                    fieldRef:
                      fieldPath: metadata.labels['tags.datadoghq.com/service']
                - name: DD_VERSION
                  valueFrom:
                    fieldRef:
                      fieldPath: metadata.labels['tags.datadoghq.com/version']
                - name: rest.endpoint.inventory
                  value: http://inventory:8084/inventory
              image: ghcr.io/steadybit/hot-deals:develop
              imagePullPolicy: Always
              livenessProbe:
                httpGet:
                  path: /actuator/health/liveness
                  port: 8099
                initialDelaySeconds: 60
                periodSeconds: 10
                timeoutSeconds: 3
              name: hot-deals
              readinessProbe:
                httpGet:
                  path: /actuator/health/readiness
                  port: 8099
                initialDelaySeconds: 60
                periodSeconds: 10
                timeoutSeconds: 3
              resources:
                limits:
                  cpu: 500m
                  ephemeral-storage: 500Mi
                  memory: 1024Mi
                requests:
                  cpu: 100m
                  ephemeral-storage: 100Mi
                  memory: 128Mi
            - image: nginx:alpine
              imagePullPolicy: Always
              livenessProbe:
                httpGet:
                  path: /actuator/health/liveness
                  port: 8083
                initialDelaySeconds: 60
                periodSeconds: 10
                timeoutSeconds: 3
              name: hot-deals-nginx
              ports:
                - containerPort: 8083
              readinessProbe:
                httpGet:
                  path: /actuator/health/readiness
                  port: 8083
                initialDelaySeconds: 60
                periodSeconds: 10
                timeoutSeconds: 5
              resources:
                limits:
                  cpu: 500m
                  ephemeral-storage: 500Mi
                  memory: 1024Mi
                requests:
                  cpu: 100m
                  ephemeral-storage: 100Mi
                  memory: 128Mi
              volumeMounts:
                - mountPath: /etc/nginx/nginx.conf
                  name: nginx-proxy-config
                  subPath: nginx.conf
          serviceAccountName: steadybit-demo
          volumes:
            - configMap:
                name: hotdeals-nginx-conf
              name: nginx-proxy-config
  3: |
    apiVersion: v1
    kind: Service
    metadata:
      labels:
        run: hot-deals
      name: hot-deals
      namespace: NAMESPACE
    spec:
      ports:
        - port: 8083
          protocol: TCP
          targetPort: 8083
      selector:
        run: hot-deals-exposed
      type: NodePort
manifest should match snapshot replica count:
  1: |
    apiVersion: v1
    data:
      nginx.conf: |-
        user  nginx;
        worker_processes  1;

        error_log  /var/log/nginx/error.log warn;
        pid        /var/run/nginx.pid;

        events {
            worker_connections  1024;
        }

        http {
            include       /etc/nginx/mime.types;
            default_type  application/octet-stream;

            access_log off;

            sendfile        on;
            keepalive_timeout  65;

            upstream hot-deals {
                server 127.0.0.1:8099;
            }

            server {
                listen 8083;

                location / {
                    proxy_pass         http://hot-deals;
                    proxy_redirect     off;
                }
            }
        }
    kind: ConfigMap
    metadata:
      name: hotdeals-nginx-conf
      namespace: NAMESPACE
  2: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        domain: shop-products
        run: hot-deals
        service-tier: "2"
        tags.datadoghq.com/service: shop-products
        tags.datadoghq.com/version: 1.0.0
        tags.steadybit.com/service-validation: http
      name: hot-deals
      namespace: NAMESPACE
    spec:
      replicas: 2
      selector:
        matchLabels:
          run: hot-deals-exposed
      strategy:
        rollingUpdate:
          maxSurge: 0
          maxUnavailable: 1
        type: RollingUpdate
      template:
        metadata:
          labels:
            domain: shop-products
            run: hot-deals-exposed
            service-tier: "2"
            tags.datadoghq.com/service: shop-products
            tags.datadoghq.com/version: 1.0.0
            tags.steadybit.com/service-validation: http
        spec:
          containers:
            - env:
                - name: SERVER_PORT
                  value: "8099"
                - name: DD_SERVICE
                  valueFrom:
                    fieldRef:
                      fieldPath: metadata.labels['tags.datadoghq.com/service']
                - name: DD_VERSION
                  valueFrom:
                    fieldRef:
                      fieldPath: metadata.labels['tags.datadoghq.com/version']
                - name: rest.endpoint.inventory
                  value: http://inventory:8084/inventory
              image: ghcr.io/steadybit/hot-deals:develop
              imagePullPolicy: Always
              livenessProbe:
                httpGet:
                  path: /actuator/health/liveness
                  port: 8099
                initialDelaySeconds: 60
                periodSeconds: 10
                timeoutSeconds: 3
              name: hot-deals
              readinessProbe:
                httpGet:
                  path: /actuator/health/readiness
                  port: 8099
                initialDelaySeconds: 60
                periodSeconds: 10
                timeoutSeconds: 3
              resources:
                limits:
                  cpu: 500m
                  ephemeral-storage: 500Mi
                  memory: 1024Mi
                requests:
                  cpu: 100m
                  ephemeral-storage: 100Mi
                  memory: 128Mi
            - image: nginx:alpine
              imagePullPolicy: Always
              livenessProbe:
                httpGet:
                  path: /actuator/health/liveness
                  port: 8083
                initialDelaySeconds: 60
                periodSeconds: 10
                timeoutSeconds: 3
              name: hot-deals-nginx
              ports:
                - containerPort: 8083
              readinessProbe:
                httpGet:
                  path: /actuator/health/readiness
                  port: 8083
                initialDelaySeconds: 60
                periodSeconds: 10
                timeoutSeconds: 5
              resources:
                limits:
                  cpu: 500m
                  ephemeral-storage: 500Mi
                  memory: 1024Mi
                requests:
                  cpu: 100m
                  ephemeral-storage: 100Mi
                  memory: 128Mi
              volumeMounts:
                - mountPath: /etc/nginx/nginx.conf
                  name: nginx-proxy-config
                  subPath: nginx.conf
          serviceAccountName: steadybit-demo
          volumes:
            - configMap:
                name: hotdeals-nginx-conf
              name: nginx-proxy-config
  3: |
    apiVersion: v1
    kind: Service
    metadata:
      labels:
        run: hot-deals
      name: hot-deals
      namespace: NAMESPACE
    spec:
      ports:
        - port: 8083
          protocol: TCP
          targetPort: 8083
      selector:
        run: hot-deals-exposed
      type: NodePort
manifest should match snapshot with podAffinity and service type and podDisruptionBudget:
  1: |
    apiVersion: v1
    data:
      nginx.conf: |-
        user  nginx;
        worker_processes  1;

        error_log  /var/log/nginx/error.log warn;
        pid        /var/run/nginx.pid;

        events {
            worker_connections  1024;
        }

        http {
            include       /etc/nginx/mime.types;
            default_type  application/octet-stream;

            access_log off;

            sendfile        on;
            keepalive_timeout  65;

            upstream hot-deals {
                server 127.0.0.1:8099;
            }

            server {
                listen 8083;

                location / {
                    proxy_pass         http://hot-deals;
                    proxy_redirect     off;
                }
            }
        }
    kind: ConfigMap
    metadata:
      name: hotdeals-nginx-conf
      namespace: NAMESPACE
  2: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        domain: shop-products
        run: hot-deals
        service-tier: "2"
        tags.datadoghq.com/service: shop-products
        tags.datadoghq.com/version: 1.0.0
        tags.steadybit.com/service-validation: http
      name: hot-deals
      namespace: NAMESPACE
    spec:
      replicas: 2
      selector:
        matchLabels:
          run: hot-deals-exposed
      strategy:
        rollingUpdate:
          maxSurge: 0
          maxUnavailable: 1
        type: RollingUpdate
      template:
        metadata:
          labels:
            domain: shop-products
            run: hot-deals-exposed
            service-tier: "2"
            tags.datadoghq.com/service: shop-products
            tags.datadoghq.com/version: 1.0.0
            tags.steadybit.com/service-validation: http
        spec:
          affinity:
            podAffinity:
              requiredDuringSchedulingIgnoredDuringExecution:
                - labelSelector:
                    matchExpressions:
                      - key: run
                        operator: In
                        values:
                          - hot-deals-exposed
                  topologyKey: topology.kubernetes.io/zone
          containers:
            - env:
                - name: SERVER_PORT
                  value: "8099"
                - name: DD_SERVICE
                  valueFrom:
                    fieldRef:
                      fieldPath: metadata.labels['tags.datadoghq.com/service']
                - name: DD_VERSION
                  valueFrom:
                    fieldRef:
                      fieldPath: metadata.labels['tags.datadoghq.com/version']
                - name: rest.endpoint.inventory
                  value: http://inventory:8084/inventory
              image: ghcr.io/steadybit/hot-deals:develop
              imagePullPolicy: Always
              livenessProbe:
                httpGet:
                  path: /actuator/health/liveness
                  port: 8099
                initialDelaySeconds: 60
                periodSeconds: 10
                timeoutSeconds: 3
              name: hot-deals
              readinessProbe:
                httpGet:
                  path: /actuator/health/readiness
                  port: 8099
                initialDelaySeconds: 60
                periodSeconds: 10
                timeoutSeconds: 3
              resources:
                limits:
                  cpu: 500m
                  ephemeral-storage: 500Mi
                  memory: 1024Mi
                requests:
                  cpu: 100m
                  ephemeral-storage: 100Mi
                  memory: 128Mi
            - image: nginx:alpine
              imagePullPolicy: Always
              livenessProbe:
                httpGet:
                  path: /actuator/health/liveness
                  port: 8083
                initialDelaySeconds: 60
                periodSeconds: 10
                timeoutSeconds: 3
              name: hot-deals-nginx
              ports:
                - containerPort: 8083
              readinessProbe:
                httpGet:
                  path: /actuator/health/readiness
                  port: 8083
                initialDelaySeconds: 60
                periodSeconds: 10
                timeoutSeconds: 5
              resources:
                limits:
                  cpu: 500m
                  ephemeral-storage: 500Mi
                  memory: 1024Mi
                requests:
                  cpu: 100m
                  ephemeral-storage: 100Mi
                  memory: 128Mi
              volumeMounts:
                - mountPath: /etc/nginx/nginx.conf
                  name: nginx-proxy-config
                  subPath: nginx.conf
          serviceAccountName: steadybit-demo
          volumes:
            - configMap:
                name: hotdeals-nginx-conf
              name: nginx-proxy-config
  3: |
    apiVersion: v1
    kind: Service
    metadata:
      labels:
        run: hot-deals
      name: hot-deals
      namespace: NAMESPACE
    spec:
      ports:
        - port: 8083
          protocol: TCP
          targetPort: 8083
      selector:
        run: hot-deals-exposed
      type: ClusterIP
  4: |
    apiVersion: policy/v1
    kind: PodDisruptionBudget
    metadata:
      name: hot-deals-pdb
      namespace: NAMESPACE
    spec:
      minAvailable: 1
      selector:
        matchLabels:
          run: hot-deals-exposed
